test/rules/prefer_typing_uninitialized_variables.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_typing_uninitialized_variables`
+
+class BadClass {
+  static var bar; // LINT
+  var foo; // LINT
+  final baz; // LINT
+  var a = 5,
+      b; // LINT
+
+  BadClass(this.baz);
+
+  void method() {
+    var bar; // LINT
+    bar = 5;
+    print(bar);
+  }
+}
+
+void aFunction() {
+  var bar; // LINT
+  bar = 5;
+  print(bar);
+}
```

Comment:
This rule will create lints for top-level variables and variables declared in for loops. There should probably be a couple of tests for those.